object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -353,24 +353,28 @@ private boolean shouldSkipInvokeCheck(Node node) {
       MethodInvocationTree methodInvokeTree = (MethodInvocationTree) callTree;
       return TreeUtils.isSuperConstructorCall(methodInvokeTree)
           || TreeUtils.isThisConstructorCall(methodInvokeTree)
-          || returnTypeIsMustCallChoiceWithOwningField((MethodInvocationNode) node)
+          || returnTypeIsMustCallChoiceWithOwningFieldOrThis((MethodInvocationNode) node)
           || hasNotOwningReturnType((MethodInvocationNode) node);
     }
     return false;
   }
 
   /**
    * Returns true if this node represents a method invocation of a must-call choice method, where
-   * the other must call choice is an owning field.
+   * the other must call choice is an owning field or this.
    *
    * @param node a method invocation node
-   * @return if this is the invocation of a method whose return type is MCC with an owning field
+   * @return if this is the invocation of a method whose return type is MCC with an owning field or
+   *     this
    */
-  private boolean returnTypeIsMustCallChoiceWithOwningField(MethodInvocationNode node) {
+  private boolean returnTypeIsMustCallChoiceWithOwningFieldOrThis(MethodInvocationNode node) {
```

Comment:
Could we rename this to `returnTypeIsMustCallChoiceWithOwningFieldOrNotOwning`?  And then indicate with a comment in the method body that for now, we assume that `this` is never owning?  No need to add any extra logic, just want to make the reasoning a bit clearer

Reply:
We could also write that if `this` is owning it could lead to false positives

Reply:
I actually renamed it to be more broad, because I realized that it doesn't matter if the field is owning - anything that's MustCallChoice with a field or `this` can be ignored.

If it's an owning field, then the class itself is responsible for closing it in its must-call method(s).

If it's a non-owning field, then this class can't be responsible - it must be a cache, or something, so we should ignore something that's MCC with that, too. I added a test case for this, too.

For `this`, I don't think it ever makes sense to have the `this` pointer be owning - that goes against the logic of must-call, which suggests that the client of a given class is responsible for calling some method on any instances of that class - the `this` pointer in a class should never be responsible for fulfilling the must-call obligations of the class.